ompi/tools/mpirun/main.c

```diff
@@ -116,6 +80,98 @@ static void setup_mca_prefixes(void)
 }
 
 
+#if OMPI_HAVE_PRTE_LAUNCH
+
+/* we can use prte_launch */
+
+int main(int argc, char *argv[])
```

Comment:
There's a lot of code duplication between this `main` and the non-prte_launch `main`.  Is there a way to consolidate / have less code duplication?

Reply:
You know, there is a "prun_common" that does pretty much what you're talking about here. I'm not sure what this "prte_launch" thing is, but just looking at the code here, it sounds just like prun_common. Might have to do a little jiggling to match signature, but not much.

Is there some reason not to use it? It was created so that "prun" and "prterun" could use the same code and eliminate duplication. Is this a case of a student not realizing it existed?